### PR TITLE
[Doc] Update README.md

### PR DESCRIPTION
**Pull request by k-ye**
Related issue = /

A major revamp of Taichi's README. To preview the full content, please go to https://github.com/k-ye/taichi/tree/readme

* More intro around what is Taichi, and its value proposition
* Cleaned up a few repeated links

I've collected and went through a dozen OSS README (see https://o9ixctz0o7.feishu.cn/docs/doccnPa9jwJ1qxdapFyYWpUrGqh#), and reshaped Taichi's README to follow this structure:

1. One-paragraph intro + key features
2. Get started
3. Documentation
4. Contacts
5. Contributing
6. Resources

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,97 +1,178 @@
...
 
-## Contributing
+The Taichi documentations are available at:
 
-We'd love to hear your comments or any of your feedback! If you would like to contribute to Taichi, please check out the [Contribution Guidelines](CONTRIBUTING.md) first.
+- [Compelete documentation](https://docs.taichi.graphics/)
+- [API reference](https://api-docs.taichi.graphics/)
 
-## Contributors
+*Due to the rapid development, both sites are still under construction. We also plan to merge them into a single site in the future.*
```
Comment (Vissidarte-Herman) [inline README.md:121]: API site: https://docs.taichi.graphics/api/

They are combined today.

File: README.md [hunk 2]
```diff
@@ -1,97 +1,178 @@
...
 
--------------------------------
+- [GitHub Issues](https://github.com/taichi-dev/taichi/issues)
+- [GitHub Discussions](https://github.com/taichi-dev/taichi/discussions)
+- [Twitter](https://twitter.com/taichigraphics)
+- [Taichi 中文论坛](https://forum.taichi.graphics/)
+- Slack & Wechat groups: please send us a message at <a href = "mailto:contact@taichi.graphics">contact@taichi.graphics</a> first, thanks!
 
-If you use Taichi in your research, please cite related papers:
+If you find any security problem, please do not hesitate to disclose it to <a href = "mailto:security@taichi.graphics?subject = Taichi Security Problem">security@taichi.graphics</a>.
```
Comment (Vissidarte-Herman) [inline README.md:133]:
```suggestion
Should you spot any security issue, do not hesitate to report it by mailing to <a href = "mailto:security@taichi.graphics?subject = Taichi Security Problem">security@taichi.graphics</a>.
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]
Minor nit-picking changes.

File: README.md (hunk 1 above)
Comment (k-ye) [reply README.md:121]: Lovely!

File: README.md [hunk 3]
```diff
@@ -1,97 +1,178 @@
...
+Advantages of using Taichi are:
 
-**Please check out our SIGGRAPH 2020 course on Taichi basics:** [YouTube](https://youtu.be/Y0-76n3aZFA), [Bilibili](https://www.bilibili.com/video/BV1kA411n7jk/), [slides (pdf)](https://yuanming.taichi.graphics/publication/2020-taichi-tutorial/taichi-tutorial.pdf).
+- Built around Python: Taichi shares the same syntax with Python, allowing you to write algorithms with minimal language barrier. It's also well integrated with the Python ecosystem, such as NumPy and PyTorch.
+- Flexibility: Taichi provides a set of generic data containers, namely *SNode* (/ˈsnoʊd/). SNode is an effective mechanism for composing hierarchical, multi-dimensional fields, unlocking many use patterns in numerical simulation (e.g. [spatially sparse computing](https://docs.taichi.graphics/lang/articles/advanced/sparse)).
+- Performance: Through the `@ti.kernel` decorator, Taichi's JIT compiler automatically parallelizes your Python functions into efficient GPU or CPU machine code.
```
[the remark this answers is not in the document]
Comment (k-ye) [reply README.md:34]: Looks good, I did `s/parallel computation/parallel execution`

File: README.md [hunk 4]
```diff
@@ -1,97 +1,178 @@
...
+**Taichi (太极)** is an open source parallel programming language for high-performance numerical computation. It is embedded in Python, and uses just-in-time (JIT) compiler frameworks (e.g. LLVM) to offload compute-intensive Python code to native GPU or CPU instructions.
 
-Advanced features of Taichi include [spatially sparse computing](https://docs.taichi.graphics/lang/articles/advanced/sparse), [differentiable programming](https://docs.taichi.graphics/lang/articles/advanced/differentiable_programming) [[examples]](https://github.com/yuanming-hu/difftaichi), and [quantized computation](https://github.com/taichi-dev/quantaichi).
+Advantages of using Taichi are:
 
-**Please check out our SIGGRAPH 2020 course on Taichi basics:** [YouTube](https://youtu.be/Y0-76n3aZFA), [Bilibili](https://www.bilibili.com/video/BV1kA411n7jk/), [slides (pdf)](https://yuanming.taichi.graphics/publication/2020-taichi-tutorial/taichi-tutorial.pdf).
+- Built around Python: Taichi shares the same syntax with Python, allowing you to write algorithms with minimal language barrier. It's also well integrated with the Python ecosystem, such as NumPy and PyTorch.
```
[the remark this answers is not in the document]
Comment (k-ye) [reply README.md]: fixed

File: README.md [hunk 5]
```diff
@@ -1,97 +1,173 @@
...
-Advanced features of Taichi include [spatially sparse computing](https://docs.taichi.graphics/lang/articles/advanced/sparse), [differentiable programming](https://docs.taichi.graphics/lang/articles/advanced/differentiable_programming) [[examples]](https://github.com/yuanming-hu/difftaichi), and [quantized computation](https://github.com/taichi-dev/quantaichi).
+Advantages of Taichi:
 
-**Please check out our SIGGRAPH 2020 course on Taichi basics:** [YouTube](https://youtu.be/Y0-76n3aZFA), [Bilibili](https://www.bilibili.com/video/BV1kA411n7jk/), [slides (pdf)](https://yuanming.taichi.graphics/publication/2020-taichi-tutorial/taichi-tutorial.pdf).
+- Built around Python: Taichi shares almost the same syntax with Python, allowing you to write algorithms with minimal language barrier. It is also well integrated into the Python ecosystem, such as NumPy and PyTorch.
+- Flexibility: Taichi provides a set of generic data containers known as *SNode* (/ˈsnoʊd/), an effective mechanism for composing hierarchical, multi-dimensional fields, and unlocking many use patterns in numerical simulation (e.g. [spatially sparse computing](https://docs.taichi.graphics/lang/articles/advanced/sparse)).
```
Comment (k-ye) [inline README.md:33]: Actually my point is that "the composed fields can unlock many use patterns", WDYT? @Vissidarte-Herman 

```suggestion
- Flexibility: Taichi provides a set of generic data containers known as *SNode* (/ˈsnoʊd/), an effective mechanism for composing hierarchical, multi-dimensional fields, which unlocks many use patterns in numerical simulation (e.g. [spatially sparse computing](https://docs.taichi.graphics/lang/articles/advanced/sparse)).
```

Comment (Vissidarte-Herman) [reply README.md:33]: hmm. Grammatically correct. The wording "unlock" does not sound about right to me. How about ", which *have* many use patterns numerical simulation xxx"

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (k-ye) [reply README.md:33]: "support" seems better to me

**Review by ailzhang — APPROVED** [review]
Awesome! Thanks!

File: README.md [hunk 6]
```diff
@@ -1,97 +1,173 @@
...
+```bash
+pip install taichi
+```
 
-### Official releases
+If you want to try out the latest features, we also provide a nightly package:
 
 ```bash
-python3 -m pip install taichi
+pip install -i https://test.pypi.org/simple/ taichi-nightly
```
[the remark this answers is not in the document]
Comment (k-ye) [reply README.md:51]: fixed

**Pull request merged**